From 27ee44fb05078102402c2b7be6ab7864063dccf6 Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Sun, 4 Jan 2015 04:06:14 +0000
Subject: [PATCH] VALIDATOR-278 - EmailValidator approves email addresses with
 hostnames starting with a dash/hyphen Fixed previously - added test to prove
 it

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/validator/trunk@1649289 13f79535-47bb-0310-9956-ffa450edef68
---
 .../commons/validator/routines/EmailValidatorTest.java       | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/test/java/org/apache/commons/validator/routines/EmailValidatorTest.java b/src/test/java/org/apache/commons/validator/routines/EmailValidatorTest.java
index 19705983f..22ef37dd2 100644
--- a/src/test/java/org/apache/commons/validator/routines/EmailValidatorTest.java
+++ b/src/test/java/org/apache/commons/validator/routines/EmailValidatorTest.java
@@ -141,6 +141,11 @@ public void testVALIDATOR_315() {
         assertTrue(validator.isValid("me@att.net")); // Make sure TLD is not the cause of the failure
     }
 
+    public void testVALIDATOR_278() {
+        assertFalse(validator.isValid("someone@-test.com"));// hostname starts with dash/hyphen
+        assertFalse(validator.isValid("someone@test-.com"));// hostname ends with dash/hyphen
+    }
+
     /**
     * Tests the email validation with commas.
     */
